var/spack/repos/builtin/packages/py-dask/package.py

```diff
@@ -138,18 +138,18 @@ class PyDask(PythonPackage):
     # Requirements for dask.distributed
     depends_on("py-dill", type=("build", "run"), when="@:0.7.5 +distributed")
     depends_on("py-pyzmq", type=("build", "run"), when="@:0.7.5 +distributed")
-    depends_on("py-distributed", type=("build", "run"), when="@0.8.2: +distributed")
-    depends_on("py-distributed@1.9:", type=("build", "run"), when="@0.9.0: +distributed")
-    depends_on("py-distributed@1.10:", type=("build", "run"), when="@0.10.0: +distributed")
-    depends_on("py-distributed@1.14:", type=("build", "run"), when="@0.12.0: +distributed")
-    depends_on("py-distributed@1.15:", type=("build", "run"), when="@0.13.0: +distributed")
-    depends_on("py-distributed@1.16:", type=("build", "run"), when="@0.14.1: +distributed")
-    depends_on("py-distributed@1.20:", type=("build", "run"), when="@0.16.0: +distributed")
-    depends_on("py-distributed@1.21:", type=("build", "run"), when="@0.17.0: +distributed")
-    depends_on("py-distributed@1.22:", type=("build", "run"), when="@0.18.0: +distributed")
-    depends_on("py-distributed@2.0:", type=("build", "run"), when="@2.0.0: +distributed")
-    depends_on("py-distributed@2020.12.0:", type=("build", "run"), when="@2020.12.0: +distributed")
-    depends_on("py-distributed@2021.6.2:", type=("build", "run"), when="@2021.6.2: +distributed")
+    depends_on("py-distributed@:2021.8.0", type=("build", "run"), when="@0.8.2: +distributed")
+    depends_on("py-distributed@1.9:2021.8.0", type=("build", "run"), when="@0.9.0: +distributed")
+    depends_on("py-distributed@1.10:2021.8.0", type=("build", "run"), when="@0.10.0: +distributed")
+    depends_on("py-distributed@1.14:2021.8.0", type=("build", "run"), when="@0.12.0: +distributed")
+    depends_on("py-distributed@1.15:2021.8.0", type=("build", "run"), when="@0.13.0: +distributed")
+    depends_on("py-distributed@1.16:2021.8.0", type=("build", "run"), when="@0.14.1: +distributed")
+    depends_on("py-distributed@1.20:2021.8.0", type=("build", "run"), when="@0.16.0: +distributed")
+    depends_on("py-distributed@1.21:2021.8.0", type=("build", "run"), when="@0.17.0: +distributed")
+    depends_on("py-distributed@1.22:2021.8.0", type=("build", "run"), when="@0.18.0: +distributed")
+    depends_on("py-distributed@2.0:2021.8.0", type=("build", "run"), when="@2.0.0: +distributed")
+    depends_on("py-distributed@2020.12.0:2021.8.0", type=("build", "run"), when="@2020.12.0: +distributed")
+    depends_on("py-distributed@2021.6.2", type=("build", "run"), when="@2021.6.2 +distributed")
```

Comment:
Dask and distributed have a weird circular dependency between them. Starting with dask 2021.?.?, the devs now pin to a specific version of distributed. If you try to use newer distributed with older dask, you get an error like below:
```
Traceback (most recent call last):
  File "/Users/ajstewart/urbanair/data/purpleair/speed.py", line 11, in <module>
    x = pd.read_csv(fpath)
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-modin-0.16.2-qsxgepq2k5b5vrqnhryljjxuedz6k2dm/lib/python3.9/site-packages/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-modin-0.16.2-qsxgepq2k5b5vrqnhryljjxuedz6k2dm/lib/python3.9/site-packages/modin/_compat/pandas_api/latest/io.py", line 156, in read_csv
    return _read(**kwargs)
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-modin-0.16.2-qsxgepq2k5b5vrqnhryljjxuedz6k2dm/lib/python3.9/site-packages/modin/_compat/pandas_api/common/io.py", line 35, in _read
    Engine.subscribe(_update_engine)
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-modin-0.16.2-qsxgepq2k5b5vrqnhryljjxuedz6k2dm/lib/python3.9/site-packages/modin/config/pubsub.py", line 217, in subscribe
    callback(cls)
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-modin-0.16.2-qsxgepq2k5b5vrqnhryljjxuedz6k2dm/lib/python3.9/site-packages/modin/pandas/__init__.py", line 159, in _update_engine
    from modin.core.execution.dask.common import initialize_dask
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-modin-0.16.2-qsxgepq2k5b5vrqnhryljjxuedz6k2dm/lib/python3.9/site-packages/modin/core/execution/dask/common/__init__.py", line 16, in <module>
    from .engine_wrapper import DaskWrapper
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-modin-0.16.2-qsxgepq2k5b5vrqnhryljjxuedz6k2dm/lib/python3.9/site-packages/modin/core/execution/dask/common/engine_wrapper.py", line 16, in <module>
    from distributed.client import default_client
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-distributed-2022.2.1-z7tfusun4h66nzcxybmk2jpmmn4xumt4/lib/python3.9/site-packages/distributed/__init__.py", line 2, in <module>
    from . import widgets  # isort:skip; load distributed widgets second
  File "/Users/ajstewart/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/py-distributed-2022.2.1-z7tfusun4h66nzcxybmk2jpmmn4xumt4/lib/python3.9/site-packages/distributed/widgets/__init__.py", line 3, in <module>
    from dask.widgets import FILTERS, TEMPLATE_PATHS
ModuleNotFoundError: No module named 'dask.widgets'
```
This `dask.widgets` import was added in distributed 2021.8.1, so 2021.8.0 is prob the newest version that works for older dask. Honestly, we could think about pinning older versions of dask to specific versions of distributed since that's what newer versions of dask do.